src/System.Text.Json/src/System/Text/Json/Serialization/JsonClassInfo.cs

```diff
@@ -16,7 +16,7 @@ namespace System.Text.Json
     internal sealed partial class JsonClassInfo
     {
         // The length of the property name embedded in the key (in bytes).
-        private const int PropertyNameKeyLength = 6;
+        private const int PropertyNameKeyLength = 7;
```

Comment:
It would be good to add some comments explaining where this number comes from and why its used.

Reply:
Sure.

Also I found that utf8 will never contain a zero byte except a single zero for NUL (UTF8 designed to be compatible with NUL terminated strings). This means we don't have to encode the length and can use the full 8 bytes.